src/Compilers/CSharp/Portable/Binder/Binder_Operators.cs

```diff
@@ -334,6 +334,7 @@ private BoundExpression BindCompoundAssignment(AssignmentExpressionSyntax node,
 
                     ReportDiagnosticsIfObsolete(diagnostics, method, node, hasBaseReceiver: false);
                     ReportDiagnosticsIfUnmanagedCallersOnly(diagnostics, method, node, isDelegateConversion: false);
+                    Debug.Assert(!IsDisallowedExtensionInOlderLangVer(method));
```

Comment:
>Debug.Assert(!IsDisallowedExtensionInOlderLangVer(method)); [](http://example.com/codeflow?start=20&length=59)

I suggest reverting this change in order to avoid unnecessary merge conflicts. This code path is specific to operators and operators are handled in a feature branch. BTW, operators must always be reported, the assert is misleading.